src/machine.rs

```diff
@@ -25,6 +22,25 @@ pub const STACK_ADDR: u64 = 32 * PAGE_SIZE; // not really about the "stack", but
 pub const STACK_SIZE: u64 = 16 * PAGE_SIZE; // whatever
 pub const NUM_CPUS: u64 = 1;
 
+pub struct FrameData<'tcx> {
+    pub call_id: stacked_borrows::CallId,
+    pub catch_panic: Option<UnwindData<'tcx>>,
+    pub is_box_me_frame: bool
+}
+
+/// Hold all of the relevant data for a call to
+/// __rust_maybe_catch_panic
```

Comment:
I think it might be a good idea to have a new file, `shims/panic.rs`, and put all of this panic-runtime-specific stuff there.